op-chain-ops/cmd/check-l2/main.go

```diff
@@ -566,6 +596,25 @@ func checkOptimismMintableERC20Factory(addr common.Address, client *ethclient.Cl
 		return err
 	}
 	log.Info("OptimismMintableERC20Factory", "_initialized", initialized)
+	if initialized.Uint64() != genesis.InitializedValue {
+		return fmt.Errorf("%w: %s", errInvalidInitialized, initialized)
+	}
+
+	abi, err := bindings.OptimismMintableERC20FactoryMetaData.GetAbi()
+	if err != nil {
+		return err
+	}
+	calldata, err := abi.Pack("initialize", common.Address{})
+	if err != nil {
+		return err
+	}
+	msg := ethereum.CallMsg{
+		To:   &addr,
+		Data: calldata,
+	}
+	if _, err := client.CallContract(context.Background(), msg, nil); err.Error() != errAlreadyInitialized.Error() {
+		return err
+	}
```

Comment:
Let's create a function that runs this check, taking an ABI as input, to dedup code.

Reply:
@Inphi We cannot do this because `initialize` takes different numbers of args but we could do something similar

Reply:
Could use variadics for a slice of `interface{}`: `func initialize(inputs ...interface{})`. It's not any worse type-safety wise as `abi.Pack` already takes an interface slice. But this is OK too.